frontend/src/lib/components/PasswordStrength.tsx

```diff
@@ -1,26 +1,58 @@
+import { LemonDivider } from '@posthog/lemon-ui'
 import { LemonProgress } from 'lib/lemon-ui/LemonProgress'
 import { Tooltip } from 'lib/lemon-ui/Tooltip'
 import zxcvbn from 'zxcvbn'
 
+export type ValidatedPasswordResult = {
+    score: number // 0 is no passsword - otherwise 1-5,
+    feedback?: string
+}
+
+export function validatePassword(password: string = ''): ValidatedPasswordResult {
+    // Checks the validation against the zxcvbn library
+    // and any other custom validation we have
+
+    const result = zxcvbn(password)
+
+    if (result.score > 3 && password.length < 8) {
+        return {
+            score: 3,
+            feedback: 'Must be at least 8 characters long',
+        }
+    }
+
+    return {
+        score: password ? result.score + 1 : 0,
+        feedback: result.feedback.suggestions.join(' '),
+    }
+}
+
 export default function PasswordStrength({
-    password = '',
-    className,
+    validatedPassword,
 }: {
-    password?: string
-    className?: string
+    validatedPassword: ValidatedPasswordResult
 }): JSX.Element {
-    // passwordScore is 0 if no password input
-    // passwordScore is 20, 40, 60, 80, or 100 if password input, based on zxcvbn score (which is 0, 1, 2, 3, or 4)
-    const passwordScore: number = password.length && zxcvbn(password).score * 20 + 20
+    const { score, feedback } = validatedPassword
 
     return (
-        <Tooltip title="Password strength">
+        <Tooltip
+            title={
+                <>
+                    Your password scores a <strong>{score}/5</strong>
```

Comment:
```suggestion
                    Your password scores a{" "}
                    <strong className="space-x-0.5">
                        <span>{score}</span>
                        <span>/</span>
                        <span>5</span>
                    </strong>
```

Just a small spacing to make it look more natural

before:
![Screenshot 2024-05-09 at 12 44 22 PM](https://github.com/PostHog/posthog/assets/6392501/f8500db6-af47-4250-b7ec-7440da5e137e)

after:
![Screenshot 2024-05-09 at 12 44 05 PM](https://github.com/PostHog/posthog/assets/6392501/80f87599-925a-4bb6-a633-c1df1eae821f)